doc(macros): document two-level macro resolution (#5371)
centreon/centreon-documentation: cloud/monitoring/basic-objects/macros.md

The **Community SNMP & Version** fields in a host form automatically generate the following custom macros:
**$_HOSTSNMPCOMMUNITY$** and **$_HOSTSNMPVERSION$**.

## Macros within macros

> When Centreon Engine evaluates a command, macro resolution happens in two successive passes.

Reviewer: ```suggestion
When Centreon Engine evaluates a command, macro resolution happens in two successive steps.
```

**Level 1 — Command level**: Macros embedded directly in the command line are resolved:

> - A resolved standard or custom macro → replaced with its value.

Reviewer: ```suggestion
- A resolved standard or custom macro is replaced with its value.
```

> - An unresolved or empty macro → replaced with an empty string.

Reviewer: ```suggestion
- An unresolved or empty macro is replaced with an empty string.
```

> **Level 2 — Macro value level**: If the value produced at Level 1 itself contains macro-like tokens, a second pass applies to that value:

Reviewer: ```suggestion
**Level 2 — Macro value level**: If the value produced at Level 1 itself contains macro-like tokens, a second step is applied to that value:
```

> - A resolved macro → replaced with its value.

Reviewer: ```suggestion
- A resolved macro is replaced with its value.
```

> - An unresolved or empty macro → **left as-is** (preserved verbatim, not stripped).

Reviewer: ```suggestion
- An unresolved or empty macro is **left as-is** (preserved verbatim, not stripped).
```

### Double-brace escape syntax

Use the `{{$MACRO$}}` format when you want unresolved macros inside a value to be replaced with an empty string rather than preserved verbatim:

- At Level 1 or Level 2: a resolved macro is replaced with its value and the double braces are removed.
- At Level 1 or Level 2: an unresolved or empty macro is replaced with an empty string and the double braces are removed.

### Use case: third-party plugin syntaxes

Some monitoring plugins use `$` characters in their own argument syntax that are not Centreon macro delimiters. A common example is NSClient++ syntax, which uses tokens such as `${name}`, `${state}`, `${problem_list}`, and `${drive}`.

Because these tokens are unresolved Centreon macros, Level 2 behaviour preserves them verbatim, allowing the plugin to receive the correct argument string.

:::warning

> If you define a macro value that contains such tokens and export the configuration with an older Engine version that does not implement the two-level model, those tokens will be stripped and the plugin will receive a malformed command line.

Reviewer: ```suggestion
> If you define a macro value that contains such tokens and export the configuration with an older Engine version that does not implement the two-level model, those tokens will be stripped and the plugin will receive a malformed command line.
```
